src/odemis/gui/comp/overlay/world.py

```diff
@@ -1231,46 +1314,350 @@ def draw(self, ctx, selected, shift=(0, 0), scale=1.0, canvas=None):
+        Left dragging on the starting point of the label allows us to change the starting point of the line and
+        change the field of interest.
+        Left dragging on the ending point of the label allows us to edit the text of the label.
+        Args:
+            drag: hover mode (HOVER_START, HOVER_END)
+            vpos: the view coordinates of the mouse cursor once left click mouse event is fired
+
+        """
+        super(LabelTool, self).start_dragging(drag, vpos, tool_mode)
+
+        if drag == gui.HOVER_START:
+            self.mode = EDIT_START_LABEL_LINE
+        elif drag == gui.HOVER_END:
+            if tool_mode == MODE_CREATE_LABEL:
+                self.mode = CREATE_LABEL
+            else:
+                self.mode = EDIT_END_LABEL_LINE
+        elif drag == gui.HOVER_TEXT:
+            self.mode = EDIT_TEXT_EDGE_MODE
+        else:
+            raise ValueError("No valid hover mode")
+
+        self._view_to_phys()
+
+    def on_motion(self, vpos, ctrl_down):
+        """
+        Given that the left mouse button is already pressed down and the mouse cursor is over the label,
+        the user can drag the label (create a new label or edit the endpoints of an existing label)
+        until the left button is released.
+        Args:
+            vpos: the view coordinates of the mouse cursor while dragging
+            ctrl_down (boolean): if True, the ctrl key is pressed while dragging and the label
+            is forced to be at one angle multiple of 45 degrees.
+        """
+        super(LabelTool, self).on_motion(vpos, ctrl_down)
+
+        if self.mode != NONE_LABEL_MODE:
+            self._update_editing(ctrl_down)
+            self.cnvs.Refresh()
+            self._view_to_phys()
+
+    def _update_editing(self, round_angle):
+        """ Update view coordinates while editing the label. If round_angle(boolean) is True,
+        the label is forced to be at one angle multiple of 45 degrees """
+        current_pos = Vec(self.cnvs.clip_to_viewport(self.drag_v_end_pos))
+
+        if self.mode == EDIT_START_LABEL_LINE:
+            if round_angle:
+                current_pos = Vec(self._round_pos(self.v_end_pos, current_pos))
+            self.v_start_pos = current_pos
+
+        elif self.mode in (EDIT_END_LABEL_LINE, CREATE_LABEL, EDIT_TEXT_EDGE_MODE):
+            if self.mode == EDIT_TEXT_EDGE_MODE:
+                self.motion = True
+            if round_angle:
+                current_pos = Vec(self._round_pos(self.v_start_pos, current_pos))
+            self.v_end_pos = current_pos
+
+    def _edit_text(self):
+        """ A dialog box pops up and the user can edit the text """
+        dlg = wx.TextEntryDialog(None, 'Enter the label', 'Text entry', value=self.text)
+
+        if dlg.ShowModal() == wx.ID_OK:
+            self.text = dlg.GetValue()
+        else:
+            logging.debug("Dialog cancelled")
+        dlg.Destroy()
+        self.calc_edges()
```

Comment:
I don't understand how this works. The `.text` has just been changed, but not yet `._label.text`, so `_label.render_pos` will be incorrect.
My approach to solve this would be:
* do not have a separate `.text`, but instead always directly set `._label.text` (or `.text` would be a short-cut to  `._label.text`).
* Change `Label`to have `render_pos` always correct. One optimised way would be to provide it via a getter, which caches the value if it's already computed.